src/csync/csync_update.cpp

```diff
@@ -648,13 +651,13 @@ int csync_ftw(CSYNC *ctx, const char *uri, csync_walker_fn fn,
   int rc = 0;
   int res = 0;
 
+  bool do_read_from_db = (ctx->current == REMOTE_REPLICA && ctx->remote.read_from_db);
+
   if (!depth) {
     mark_current_item_ignored(ctx, previous_fs, CSYNC_STATUS_INDIVIDUAL_TOO_DEEP);
     goto done;
   }
 
-  bool do_read_from_db = (ctx->current == REMOTE_REPLICA && ctx->remote.read_from_db);
```

Comment:
Just commenting to make you aware of it..
I guess this reordering has no effect..

Reply:
C++ complains if a goto crosses a variable declaration.